File: 006 CS Inheritance/MyApp.cs
// 006 C# Héritage
// Essais d'héritage simple en C#
// 2001-01-27   PV
// 2006-10-01   PV  VS2005
// 2012-02-25   PV  VS2010


using System;

abstract class B
{
    protected B()
    {
        Console.WriteLine("B.ctor");
    }

    protected virtual void SomeVirtualMethod()
    {

    }

    protected abstract void AnAbstractMehod();

    protected void NonVirtualMethod()
    {

    }
}

class D : B
{
    public D()
    {
        Console.WriteLine("D.ctor");
    }

    protected override void SomeVirtualMethod()
    {

    }

    protected override void AnAbstractMehod()
    {

    }

    protected new void NonVirtualMethod()
    {

    }
}


class MyApp
{
    public static void Main()
    {
        D d = new D();
        Console.ReadLine();
    }
}

File: Net6Converted 001-019/009 VB Inheritance/MyApp.vb
' 09 VB héritage
' Essais d'héritage de classes en VB
' 2001-10-27    PV
' 2006-10-01    PV  VS2005
' 2012-02-25	PV	VS2010
' 2021-09-17    PV  VS2022/Net6

Module Module1

    Sub Main()
        Dim d As New D()
        Console.ReadLine()
    End Sub

End Module

Class B

    Sub New()
        Console.WriteLine("B.ctor")
    End Sub

End Class

Class D
    Inherits B

    Sub New()
        'MyBase.New()   'Inutile, appelé automatiquement
        Console.WriteLine("D.ctor")
    End Sub

End Class